src/libexpr/value.hh

```diff
@@ -29,6 +29,22 @@ typedef enum {
     tFloat
 } ValueType;
 
+// This type abstracts over all actual value types in the language,
+// grouping together implementation details like tList*, different function
+// types, and types in non-normal form (so thunks and co.)
+typedef enum {
+    nThunk,
```

Comment:
`NormalType` might be a bit of a confusing name since thunks are not normal forms. Maybe `AbstractType`?

Reply:
How about this: We make `ValueType type` be an unnamed union with
```c++
union {
  tInt = 1,
  // ...
} type;
```

And then let's use the `ValueType` name for the current `NormalType`.

Reply:
Then we should probably also use `Value::type()` instead of `Value::normalType()`

Reply:
Yeah that sounds good.

Reply:
Wasn't able to easily use an anonymous enum because the `showType` and `printValue` friends wouldn't have access to the variants like that. Instead I renamed the original `ValueType` to `InternalType`, so that `NormalType` can be `ValueType` instead. Also renamed the field to `Value::internalType` which is needed for `Value::type()` to be a function.